### PR TITLE
Login/setup new site with bitbucket

### LINKED ISSUE #39: Login with / connect to Bitbucket
* [x] Login
* [x] Setup site config from list of repos
* [x] Auto-set-up webhook
* [ ] Auto set-up deploy key (can we use deploy key? we need push access)

### PR DESCRIPTION
**Pull request by ericroberts**
Closes #39

### CODE REVIEW

File: app/controllers/site.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@
 require_relative "../git_providers/git_providers"
 require_relative "../git_providers/github"
 require_relative "../git_providers/gitlab"
+require_relative "../git_providers/bitbucket"
```
Comment (singpolyma) [inline app/controllers/site.rb]: does not look alphabetical

File: app/git_providers/bitbucket.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
+require "json"
+require "httparty"
```
Comment (singpolyma) [inline app/git_providers/bitbucket.rb]: should go in gemfile if we're going to use this

File: app/git_providers/bitbucket.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+				user, repo = extract_from_remote(remote)
+				self.class.post(
+					"/2.0/repositories/#{user}/#{repo}/hooks",
+					body: {
+						url: callback,
+						active: true,
+						events: ["repo:push"]
+					}.to_json,
+					headers: {
+						"Authorization" => "Bearer #{omniauth["credentials"]["token"]}",
```
Comment (singpolyma) [inline app/git_providers/bitbucket.rb]: try refactor this with HTTParty features

**Comment by singpolyma** [comment]
can't use deploy key, add to user like with gitlab

File: app/git_providers/bitbucket.rb (hunk 2 above)
Comment (ericroberts) [inline app/git_providers/bitbucket.rb]: done

**Pull request merged**